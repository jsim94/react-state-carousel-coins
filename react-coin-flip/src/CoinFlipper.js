import React, { useState } from "react";
import Coin from "./Coin";

function CoinFlipper() {
  const [side, setSide] = useState();
  const [flips, setFlips] = useState(0);
  const [heads, setHeads] = useState(0);
  const [tails, setTails] = useState(0);

  function flip() {
    const randBool = Math.random() > 0.5 ? true : false;

    setSide(randBool);
    setFlips(flips + 1);

    if (randBool) {
      setHeads(heads + 1);
    } else {
      setTails(tails + 1);
    }
  }

  return (
    <div className="CoinFlipper">
      <Coin side={side} />
      <button onClick={flip} data-testid="flip-button">
        FLIP MEEEE
      </button>
      <p>
        Out of {flips} flips, there have been {heads} heads(green) and {tails} tails(red).
      </p>
    </div>
  );
}

export default CoinFlipper;
