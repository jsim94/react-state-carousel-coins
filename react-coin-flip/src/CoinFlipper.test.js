import React from "react";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import CoinFlipper from "./CoinFlipper";

it("should render", function () {
  render(<CoinFlipper />);
});

it("should match snapshot", function () {
  const { asFragment } = render(<CoinFlipper />);
  expect(asFragment()).toMatchSnapshot();
});

describe("Pressing flip button", () => {
  beforeEach(() => {
    jest.spyOn(Math, "random").mockReturnValueOnce(0.25).mockReturnValueOnce(0.75);
  });

  afterEach(function () {
    Math.random.mockRestore();
  });

  it("should return tails then heads", () => {
    const { getByTestId } = render(<CoinFlipper />);

    const coin = getByTestId("coin");
    const flipButton = getByTestId("flip-button");

    fireEvent.click(flipButton);
    expect(coin).toHaveStyle({ background: "red" });

    fireEvent.click(flipButton);
    expect(coin).toHaveStyle({ background: "green" });
  });
});
