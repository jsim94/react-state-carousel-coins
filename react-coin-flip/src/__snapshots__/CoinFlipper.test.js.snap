// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="CoinFlipper"
  >
    <div
      class="Coin"
      data-testid="coin"
      style="background: red;"
    />
    <button
      data-testid="flip-button"
    >
      FLIP MEEEE
    </button>
    <p>
      Out of 0 flips, there have been 0 heads(green) and 0 tails(red).
    </p>
  </div>
</DocumentFragment>
`;
